scaletest/terraform/action/coder_templates.tf

```diff
@@ -1,109 +1,285 @@
+    affinity {
+      node_affinity {
+        required_during_scheduling_ignored_during_execution {
+          node_selector_term {
+            match_expressions {
+              key = "cloud.google.com/gke-nodepool"
+              operator = "In"
+              values = ["${google_container_node_pool.node_pool["primary_workspaces"].name}","${google_container_node_pool.node_pool["europe_workspaces"].name}","${google_container_node_pool.node_pool["asia_workspaces"].name}"]
+            }
+          }
         }
       }
+    }
+  }
+}
+EOF
+}
+
+resource "kubernetes_config_map" "template_primary" {
+  provider = kubernetes.primary
+
+  metadata {
+    name      = "coder-template"
+    namespace = kubernetes_namespace.coder_primary.metadata.0.name
+  }
+
+  data = {
+    "main.tf" = local_file.kubernetes_template.content
+  }
+}
+
+resource "kubernetes_job" "push_template_primary" {
+  provider = kubernetes.primary
+
+  metadata {
+    name      = "${var.name}-push-template"
+    namespace = kubernetes_namespace.coder_primary.metadata.0.name
+    labels = {
+      "app.kubernetes.io/name" = "${var.name}-push-template"
+    }
+  }
+  spec {
+    completions = 1
+    template {
+      metadata {}
       spec {
-        security_context {
-          run_as_user = "1000"
-          fs_group    = "1000"
+        affinity {
+          node_affinity {
+            required_during_scheduling_ignored_during_execution {
+              node_selector_term {
+                match_expressions {
+                  key      = "cloud.google.com/gke-nodepool"
+                  operator = "In"
+                  values   = ["${google_container_node_pool.node_pool["primary_misc"].name}"]
+                }
+              }
+            }
+          }
         }
         container {
-          name              = "dev"
-          image             = "${var.workspace_image}"
-          image_pull_policy = "Always"
-          command           = ["sh", "-c", coder_agent.main.init_script]
-          security_context {
-            run_as_user = "1000"
-          }
-          env {
-            name  = "CODER_AGENT_TOKEN"
-            value = coder_agent.main.token
+          name  = "cli"
+          image = "${var.coder_image_repo}:${var.coder_image_tag}"
+          command = [
+            "/opt/coder",
+            "--verbose",
+            "--url=${local.deployments.primary.url}",
+            "--token=${trimspace(data.local_file.api_key.content)}",
+            "templates",
+            "push",
+            "--directory=/home/coder/template",
+            "--provisioner-tag=scope=organization",
+            "--provisioner-tag=deployment=primary",
+            "--yes",
+            "kubernetes-primary"
+          ]
+          volume_mount {
+            name       = "coder-template"
+            mount_path = "/home/coder/template/main.tf"
+            sub_path   = "main.tf"
```

Comment:
future enhancement idea: could we use the `coderd` Terrraform provider for this?

Reply:
`coderd` provider can't be used here because it needs a valid coder token up front, and we don't have one since we create coder in the same apply. We've discussed plans to make this possible in the future but it's currently not possible with the provider.